lms/djangoapps/ccx/migrations/0004_seed_forum_roles_in_ccx_courses.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from ccx_keys.locator import CCXLocator
+from courseware.courses import get_course_by_id
+from django.db import migrations
+from django.http import Http404
+
+from django_comment_common.models import FORUM_ROLE_ADMINISTRATOR, assign_role
+from django_comment_common.utils import seed_permissions_roles
+
+def seed_forum_roles_for_existing_ccx(apps, schema_editor):
+    """
+    Seed forum roles and make CCX coach forum admin for respective CCX(s).
+
+    Arguments:
+        apps (Applications): Apps in edX platform.
+        schema_editor (SchemaEditor): For editing database schema i.e create, delete field (column)
+
+    """
+    CustomCourseForEdX = apps.get_model("ccx", "CustomCourseForEdX")
+    Role = apps.get_model("django_comment_common", "Role")
+    list_ccx = CustomCourseForEdX.objects.all()
+    for ccx in list_ccx:
+        if not ccx.course_id or ccx.course_id.deprecated:
+            # prevent migration for deprecated course ids or invalid ids.
+            continue
+        ccx_locator = CCXLocator.from_course_locator(ccx.course_id, unicode(ccx.id))
+        # Seed forum roles.
+        seed_permissions_roles(ccx_locator)
+        # Make CCX coach forum admin.
+        admin_role = Role.objects.get(course_id=ccx_locator, name=FORUM_ROLE_ADMINISTRATOR)
+        ccx.coach.roles.add(admin_role)
+
+
+def remove_forum_roles_for_existing_ccx(apps, schema_editor):
+    """
+    Remove seeded forum roles from respective CCX(s).
+
+    Arguments:
+        apps (Applications): Apps in edX platform.
+        schema_editor (SchemaEditor): For editing database schema i.e create, delete field (column)
+
+    """
+    CustomCourseForEdX = apps.get_model("ccx", "CustomCourseForEdX")
+    Role = apps.get_model("django_comment_common", "Role")
+    list_ccx = CustomCourseForEdX.objects.all()
+    for ccx in list_ccx:
+        if not ccx.course_id or ccx.course_id.deprecated:
+            # prevent migration for deprecated course ids or invalid ids.
+            continue
+        ccx_locator = CCXLocator.from_course_locator(ccx.course_id, unicode(ccx.id))
+        roles = Role.objects.filter(course_id=ccx_locator)
+        roles.delete()
```

Comment:
The roles would not have existed prior to the forward migration being run because discussion forums were not enabled on CCX courses before the changes in this PR at all.

However, leaving roles and permissions in the database won't do any harm, so I just removed the reverse migration for the sake of simplicity.